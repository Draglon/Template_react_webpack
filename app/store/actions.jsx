export const POPUP_TOGGLE = 'POPUP_TOGGLE';
export const MENU_TOGGLE = 'MENU_TOGGLE';

export const popupToggle = () => {
    return {
        type: POPUP_TOGGLE
    };
};

export const menuToggle = () => {
    return {
        type: MENU_TOGGLE
    };
};