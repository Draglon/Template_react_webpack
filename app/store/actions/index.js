export * from './ui';
export * from './login';
