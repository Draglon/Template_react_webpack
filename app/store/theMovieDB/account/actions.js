import * as t from './actionTypes';

export const profileRequest = () => ({ type: t.PROFILE_REQUEST });
export const profileSuccess = payload => ({ type: t.PROFILE_SUCCESS, payload });
export const profileFailure = payload => ({ type: t.PROFILE_FAILURE, payload });
