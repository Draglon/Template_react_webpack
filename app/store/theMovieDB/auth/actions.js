import * as t from './actionTypes';

export const createSessionRequest = payload => ({ type: t.CREATE_SESSION_REQUEST, payload });
export const createSessionSuccess = payload => ({ type: t.CREATE_SESSION_SUCCESS, payload });
export const createSessionFailure = payload => ({ type: t.CREATE_SESSION_FAILURE, payload });

export const deleteSessionRequest = () => ({ type: t.DELETE_SESSION_REQUEST });
export const deleteSessionSuccess = payload => ({ type: t.DELETE_SESSION_SUCCESS, payload });
export const deleteSessionFailure = payload => ({ type: t.DELETE_SESSION_FAILURE, payload });
