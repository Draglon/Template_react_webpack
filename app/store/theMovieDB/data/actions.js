import * as t from './actionTypes';

export const addEntities = entities => ({ type: t.ADD_ENTITIES, entities });
