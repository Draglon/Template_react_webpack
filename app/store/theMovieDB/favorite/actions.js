import * as t from './actionTypes';

export const favoriteRequest = payload => ({ type: t.FAVORITE_REQUEST, payload });
export const favoriteSuccess = payload => ({ type: t.FAVORITE_SUCCESS, payload });
export const favoriteFailure = payload => ({ type: t.FAVORITE_FAILURE, payload });

export const addToFavoriteRequest = payload => ({ type: t.ADD_TO_FAVORITE_REQUEST, payload });
export const addToFavoriteSuccess = payload => ({ type: t.ADD_TO_FAVORITE_SUCCESS, payload });
export const addToFavoriteFailure = payload => ({ type: t.ADD_TO_FAVORITE_FAILURE, payload });
