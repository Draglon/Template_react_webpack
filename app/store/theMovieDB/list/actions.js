import * as t from './actionTypes';

// Create list
export const createListRequest = payload => ({ type: t.CREATE_LIST_REQUEST, payload });
export const createListSuccess = payload => ({ type: t.CREATE_LIST_SUCCESS, payload });
export const createListFailure = payload => ({ type: t.CREATE_LIST_FAILURE, payload });

// Add movie
// export const addMovieRequest = payload => ({ type: t.ADD_MOVIE_REQUEST, payload });
// export const addMovieSuccess = payload => ({ type: t.ADD_MOVIE_SUCCESS, payload });
// export const addMovieFailure = payload => ({ type: t.ADD_MOVIE_FAILURE, payload });
