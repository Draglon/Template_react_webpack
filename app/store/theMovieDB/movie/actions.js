import * as t from './actionTypes';

export const movieRequest = payload => ({ type: t.MOVIE_REQUEST, payload });
export const movieFailure = payload => ({ type: t.MOVIE_FAILURE, payload });
