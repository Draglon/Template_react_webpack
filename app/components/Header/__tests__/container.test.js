import React from 'react';
import configureStore from 'redux-mock-store';

import { profileRequest } from '../../../store/theMovieDB/profile/actions';
import HeaderContainer from '../container';

describe('<HeaderContainer />', () => {
  const id = 8493834;
  const profileData = {
    id,
    avatar: 'some hash',
    name: 'some name',
    username: 'some username',
  };
  const state = {
    reducers: {
      profile: {
        data: {
          id,
        },
      },
      data: {
        profile: {
          id,
        },
      },
    },
  };
  const props = {
    profile: profileData,
  };

  const store = configureStore()(state);
  store.dispatch = jest.fn();

  const wrapper = shallow(<HeaderContainer store={store} {...props} />);
  const container = wrapper.dive();
  const instance = container.instance();

  describe('componentDidMount()', () => {
    it('dispatches the profileRequest()', () => {
      instance.componentDidMount();

      expect(store.dispatch).toHaveBeenCalledWith(profileRequest());
    });
  });

  describe('Render snapshot', () => {
    it('HeaderContainer', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
