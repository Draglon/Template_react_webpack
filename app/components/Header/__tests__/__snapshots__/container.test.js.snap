// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderContainer /> Render snapshot HeaderContainer 1`] = `
<HeaderContainer
  profileRequest={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "type": "PROFILE_REQUEST",
            },
          ],
          Array [
            Object {
              "type": "PROFILE_REQUEST",
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "type": "PROFILE_REQUEST",
              },
            ],
            Array [
              Object {
                "type": "PROFILE_REQUEST",
              },
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
